### PR TITLE
Format code with gofumpt

### PR DESCRIPTION
**Pull request by thaJeztah**
Opening as draft, as we should also add CI/validation for this if we want to use this. (also see https://github.com/opencontainers/runc/pull/2975)

This formats the code using https://github.com/mvdan/gofumpt, which provides a superset of the default `gofmt`, but with some stricter (but sensible) rules.

This PR commits the results of;

```bash
git ls-files \*.go | grep -v ^vendor/ | grep -v 'pb.go$' | xargs gofumpt -w
```

(but chunked ~ by package)

### CODE REVIEW

**Comment by thaJeztah** [comment]
Failure on Windows is unrelated (https://github.com/moby/moby/issues/39352);

```
=== RUN   TestDockerSuite/TestRunInteractiveWithRestartPolicy
    docker_cli_run_test.go:1794: assertion failed: 
        Command:  d:\CI\PR-43169\1\binary\docker.exe run -i --name test-inter-restart --restart=always busybox sh
        ExitCode: 0
        Stdout:   
        Stderr:   
        
        Failures:
        ExitCode was 0 expected 11
    --- FAIL: TestDockerSuite/TestRunInteractiveWithRestartPolicy (6.52s)
```

**Review by tonistiigi — APPROVED** [review]
SGTM for a one-off update. The CI should continue to only validate against upstream `gofmt` though.

I hope the rebase hell this causes is worth it.

File: api/server/router/swarm/backend.go
Comment (AkihiroSuda) [inline api/server/router/swarm/backend.go:42]: We may want to keep these empty lines for readability.

**Review by AkihiroSuda — APPROVED** [review]

File: api/server/router/swarm/backend.go
Comment (thaJeztah) [reply api/server/router/swarm/backend.go:42]: Yes, I was looking at some of those. I think in such cases we should either;

- Add comments in between (describing what "sections" there are)
- Perhaps better: define smaller interfaces (and `Backend` could embed the smaller interfaces), e.g. `NodeBackend`, `SecretsBackend`, `ConfigBackend`)

**Comment by thaJeztah** [comment]
FTR; so (as Tonis mentioned above), we discussed this in the maintainers-meeting, and consensus was that;

- most (all?) people present agreed that the `gofumpt` formatting is "good" (and that it should really be included in the default `go fmt`)
- manually formatting the code (periodically) with `gofumpt` is ok
- .. but _enforcing_ this format was not desirable at this point;
    - most users won't have `gofumpt` installed
    - they may have their editor / IDE set up to use `gofmt` / `go fmt`
    - if they would change their editor's settings to use `gofumpt`, this could lead to the same formatting to be applied to "any" project they're working on (most of those would not be using `gofumpt` currently)


So we decided to periodically do a `gofumpt` of the codebase (hoping that `gofumpt` will be rolled into `go fmt` at some point 🤞)

**Comment by thaJeztah** [comment]
> I hope the rebase hell this causes is worth it.

I'm good with holding off merging this PR for a bit (there's quite some PRs pending, and now that Windows CI is "fixed", I think we should try to get some of those merged); happy to rebase this one a couple of times if needed

**Comment by cpuguy83** [comment]
I'd just rather not do this because it is just going to cause periodic rebase hell.

**Review by neersighted — APPROVED** [review]

**Pull request merged**
